src/main/java/org/folio/circulation/domain/OverdueFineCalculatorService.java

```diff
@@ -0,0 +1,201 @@
+    FeeFineOwner feeFineOwner;
+    FeeFine feeFine;
+
+    CalculationParameters(Loan loan, Item item, FeeFineOwner feeFineOwner, FeeFine feeFine) {
+      this.loan = loan;
+      this.item = item;
+      this.feeFineOwner = feeFineOwner;
+      this.feeFine = feeFine;
+    }
+
+    CalculationParameters withItem(Item item) {
+      return new CalculationParameters(this.loan, item, this.feeFineOwner, this.feeFine);
+    }
+
+    CalculationParameters withOwner(FeeFineOwner feeFineOwner) {
+      return new CalculationParameters(this.loan, this.item, feeFineOwner, this.feeFine);
+    }
+
+    CalculationParameters withFeeFine(FeeFine feeFine) {
+      return new CalculationParameters(this.loan, this.item, this.feeFineOwner, feeFine);
+    }
+  }
+
+  private static final Map<OverdueFineInterval, Integer> MINUTES_IN_INTERVAL =
+    new EnumMap<>(OverdueFineInterval.class);
+  static {
+    MINUTES_IN_INTERVAL.put(OverdueFineInterval.MINUTE, 1);
+    MINUTES_IN_INTERVAL.put(OverdueFineInterval.HOUR, 60);
+    MINUTES_IN_INTERVAL.put(OverdueFineInterval.DAY, 1440);
+    MINUTES_IN_INTERVAL.put(OverdueFineInterval.WEEK, 10080);
+    MINUTES_IN_INTERVAL.put(OverdueFineInterval.MONTH, 44640);
+    MINUTES_IN_INTERVAL.put(OverdueFineInterval.YEAR, 525600);
+  }
+
+  public static OverdueFineCalculatorService using(Clients clients) {
+    return new OverdueFineCalculatorService(clients);
+  }
+
+  private OverdueFinePolicyRepository overdueFinePolicyRepository;
+  private AccountRepository accountRepository;
+  private ItemRepository itemRepository;
+  private FeeFineOwnerRepository feeFineOwnerRepository;
+  private FeeFineRepository feeFineRepository;
+  private OverduePeriodCalculatorService overduePeriodCalculatorService;
+
+  public OverdueFineCalculatorService(
+    OverdueFinePolicyRepository overdueFinePolicyRepository,
+    AccountRepository accountRepository,
+    ItemRepository itemRepository,
+    FeeFineOwnerRepository feeFineOwnerRepository,
+    FeeFineRepository feeFineRepository,
+    OverduePeriodCalculatorService overduePeriodCalculatorService) {
+    this.overdueFinePolicyRepository = overdueFinePolicyRepository;
+    this.accountRepository = accountRepository;
+    this.itemRepository = itemRepository;
+    this.feeFineOwnerRepository = feeFineOwnerRepository;
+    this.feeFineRepository = feeFineRepository;
+    this.overduePeriodCalculatorService = overduePeriodCalculatorService;
+  }
+
+  private OverdueFineCalculatorService(Clients clients) {
+    this(new OverdueFinePolicyRepository(clients),
+      new AccountRepository(clients),
+      new ItemRepository(clients, true, false, false),
+      new FeeFineOwnerRepository(clients),
+      new FeeFineRepository(clients),
+      new OverduePeriodCalculatorService(new CalendarRepository(clients)));
+  }
+
+  public CompletableFuture<Result<CheckInProcessRecords>> calculateOverdueFine(
+    CheckInProcessRecords records) {
+    Loan loan = records.getLoan();
+    if (loan == null) {
```

Comment:
Personally, I would introduce whitespace between the multiline method signature, the variable and the if branch statement

Reply:
Done